src/main/java/com/uw/dao/TrainerDaoImpl.java

```diff
@@ -1,38 +1,114 @@
 package com.uw.dao;
 
 import com.uw.model.Trainer;
-import com.uw.util.Storage;
+import com.uw.model.User;
+import jakarta.persistence.Query;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 public class TrainerDaoImpl implements TrainerDao{
 
-    private Storage storage;
+    private SessionFactory sessionFactory;
 
     @Autowired
-    public void setStorage(Storage storage){
-        this.storage = storage;
+    public void setSessionFactory(SessionFactory sessionFactory){
+        this.sessionFactory = sessionFactory;
     }
 
     @Override
     public void create(Trainer trainer) {
-        long id = this.storage.getSize() + 1;
-        trainer.setUserId(id);
-        this.storage.save( trainer.getUserId(), trainer );
+        Session session = sessionFactory.openSession();
+        Transaction transaction = null;
+        try {
+            transaction = session.beginTransaction();
+            session.persist(trainer);
+            transaction.commit();
+        } catch ( Exception e ){
+            transaction.rollback();
+        }finally{
+            session.close();
+        }
     }
 
     @Override
     public void update(Trainer trainer) {
-        this.storage.save( trainer.getUserId(), trainer );
+        Session session = sessionFactory.openSession();
+        Transaction transaction = null;
+        try {
+            transaction = session.beginTransaction();
+            session.merge(trainer);
+            transaction.commit();
+        } catch ( Exception e ){
+            transaction.rollback();
+        }finally {
+            session.close();
+        }
     }
 
     @Override
     public Trainer selectProfile(long id) {
-        return (Trainer) this.storage.get( id );
+        Session session = sessionFactory.openSession();
+        Trainer trainer = null;
+        try{
+            trainer = session.get(Trainer.class, id);
+        }catch ( Exception e ){
+            e.printStackTrace();
+        }finally{
+            session.close();
+        }
+        return trainer;
+    }
+
+    @Override
+    public List<Trainer> findAll() {
+        Session session = sessionFactory.openSession();
+        List<Trainer> trainerList = null;
```

Comment:
avoid null initialization instead of that you could use List.of() to get an empty one